src/decomp_2d.f90

```diff
@@ -74,6 +74,28 @@ module decomp_2d
   ! flags for periodic condition in three dimensions
   logical, save :: periodic_x, periodic_y, periodic_z
 
+  !
+  ! Output for the log can be changed by the external code before calling decomp_2d_init
+  !
+  !    0 => No log output
+  !    1 => Master rank log output to stdout
+  !    2 => Master rank log output to the file "decomp_2d_setup.log"
+  !    3 => All ranks log output to a dedicated file
+  !
+  ! The default value is 2 (3 for debug builds)
+  !
+  enum, bind(c)
+     enumerator :: D2D_LOG_NO = 0
+     enumerator :: D2D_LOG_STDOUT = 1
+     enumerator :: D2D_LOG_FILE = 2
+     enumerator :: D2D_LOG_ALL = 3
+  end enum
```

Comment:
Can the user pass these values, e.g. `D2D_LOG_NO` rather than `0` to `decomp_2d_init`? I think this would be better if possible and should be the recommendation in the README too if so.

Reply:
This is not possible with the enum structure. But the code could export a list of parameters, such as the ones above yes. Similarly to what FFTW does, it is probably even possible to generate this list of parameters automatically.

Reply:
The enum was removed in 333ad0b714e361fb57d141b37ae84d4b435b3c64. The code is now easier to call from Fortran and harder to call from C.